pkg/controller/hibernation/hibernation_controller.go

```diff
@@ -482,7 +484,7 @@ func (r *hibernationReconciler) checkClusterStopped(cd *hivev1.ClusterDeployment
 }
 
 func (r *hibernationReconciler) checkClusterRunning(cd *hivev1.ClusterDeployment, logger log.FieldLogger,
-	readyCondition *hivev1.ClusterDeploymentCondition, hibernatingCondition *hivev1.ClusterDeploymentCondition) (reconcile.Result, error) {
+	readyCondition *hivev1.ClusterDeploymentCondition) (reconcile.Result, error) {
```

Comment:
Could the `readyCondition` parameter here suffer from the same issue?

Reply:
I see that the `reason` of the condition is checked and this is why we can ensure we're dealing with the right one.

Reply:
Nope, it's just by luck that no path to this point runs through `setCDCondition` without immediately returning.